test/test_nn.py

```diff
@@ -1075,6 +1075,17 @@ def test_AvgPool3d_backward_after_cat_dim1_cuda(self):
 
         y.backward(grad)
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    def test_conv_transpose2d_contig_wrong_stride_dim1_cuda(self):
+        # x has to have batch_size 1 to test contiguous checks
+        x = torch.randn(1, 16, 5, 5).cuda()
+        stride = list(x.stride())
+        stride[0] = 20
+        # change the stride in dimension 0. the tensor is still contiguous because size[0] is 1
+        x.set_(x.storage(), 0, x.size(), stride)
+        assert x.is_contiguous()
+        F.conv_transpose2d(Variable(x), Variable(torch.randn(16, 1, 1, 1)).cuda())
```

Comment:
Do we really need to run Conv2d for this test? Maybe just add one for contiguous() and ensure that it fixes the strides. It’s much simpler, faster and better describes the problem. Please move it to test_torch.py